src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+
+export const useAppSelector = useSelector.withTypes<RootState>();
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
```

Comment:
The `useDispatch.withTypes<AppDispatch>()` is incorrect. You should use `useDispatch` with TypeScript generics like this: `export const useAppDispatch = () => useDispatch<AppDispatch>();`.